pkg/commands/docker.go

```diff
@@ -160,19 +160,22 @@ func PopulateDigests(runner build.Runner, dependencies []build.ImageDependencies
 }
 
 func queryDigest(runner build.Runner, reference *build.ImageReference) error {
-	refString := reference.String()
-	output, err := runner.QueryCmd("docker", []string{
-		"inspect", "--format", "\"{{json .RepoDigests}}\"", refString,
-	})
-	if err != nil {
-		return err
-	}
+	if reference != nil {
+		refString := reference.String()
+		output, err := runner.QueryCmd("docker", []string{
+			"inspect", "--format", "\"{{json .RepoDigests}}\"", refString,
+		})
+		if err != nil {
+			return err
+		}
+
+		trimCharPredicate := func(c rune) bool {
```

Comment:
can you refactor this to be real func and pass it in instead

Reply:
For now, I actually prefer to keep it as an anonymous in the function scope as it's very specific to the digest case (eg, it trim double quote). I also want to keep it close to the caller to make it easy to find. We can also refactor is later once we need to reuse the logic =)